pkg/cluster/internal/providers/docker/provision.go

```diff
@@ -64,11 +65,13 @@ func planCreation(cluster string, cfg *config.Cluster) (createContainerFuncs []f
 		// fixup relative paths, docker can only handle absolute paths
 		for i := range node.ExtraMounts {
 			hostPath := node.ExtraMounts[i].HostPath
-			absHostPath, err := filepath.Abs(hostPath)
-			if err != nil {
-				return nil, errors.Wrapf(err, "unable to resolve absolute path for hostPath: %q", hostPath)
+			if !fs.IsAbs(hostPath) {
+				absHostPath, err := filepath.Abs(hostPath)
+				if err != nil {
+					return nil, errors.Wrapf(err, "unable to resolve absolute path for hostPath: %q", hostPath)
+				}
+				node.ExtraMounts[i].HostPath = absHostPath
```

Comment:
should be outside the if?

Reply:
pseudo-code of the current logic:
```
if not absolute(path):
  make_absolute(path)
```
so no?

Reply:
why not detecting the the OS?
`if runtime.GOOS == "windows" `

Reply:
oh right, nvm

Reply:
the behavior is correct on linux with the current very simple implementation

remote hosts make this tricky and will need a more elaborate logic, which I'm mulling, but the priority is:
- preserve the existing behavior generally
- fix the case of posix absolute path on windows

this is the minimum implementation of that and pretty simple / cheap in the common case, on linux it just checks the first character at most twice (in the case that it is relative)

Reply:
remote hosts (even locally remote, think WSL2) make this a bit more complex, I think we can solve it but I wanted to go ahead and see if we could sort out the original issue cheaply and get that verified